src/GraphQL/Types/Composite/AutoRegisteringOutputHelper.cs

```diff
@@ -70,13 +69,12 @@ internal static class AutoRegisteringOutputHelper
                 {
                     var resolver = new SourceStreamMethodResolver(methodInfo, memberInstanceExpression, expressions);
                     fieldType.Resolver = resolver;
-                    fieldType.StreamResolver = resolver;
+                    (fieldType as SubscriptionRootFieldType ?? throw new InvalidOperationException("BOOM!")).StreamResolver = resolver;
```

Comment:
@Shane32 What would you like to do here?

Reply:
`AutoRegisteringHelper.CreateField` creates the `FieldType`, but this is entirely separate from the this code that builds the resolvers.  We would have to cross the logic between the two, so `AutoRegisteringHelper.CreateField` calls `AutoRegisteringOutputHelper.IsObservableOrAsyncEnumerable` to determine what kind of `FieldType` to add.

Reply:
Which it looks like you've already done.

Reply:
I mean would you like to throw exception or not. If yes then which type and message?

Reply:
I suggest either:

1. `new ArgumentException("The member is a stream resolver, but the specified fieldType is not a SubscriptionRootFieldType.", nameof(fieldType));`
2. `new InvalidOperationException("The member is a stream resolver, but the specified fieldType is not a SubscriptionRootFieldType.");`